pkg/apis/batch/validation/validation.go

```diff
@@ -90,6 +90,12 @@ func ValidateGeneratedSelector(obj *batch.Job) field.ErrorList {
 		allErrs = append(allErrs, field.Required(field.NewPath("metadata").Child("uid"), ""))
 	}
 
+	controllerUidLegacy := "controller-uid"
+	jobNameLegacy := "job-name"
+	// Use Label Prefix for controller-uid and job name
+	controllerUid := batch.JobLabelPrefix + "controller-uid"
```

Comment:
they could be package constants themselves

Reply:
done.